utils_linux.go

```diff
@@ -281,6 +281,11 @@ func (r *runner) run(config *specs.Process) (int, error) {
 	}
 	baseFd := 3 + len(process.ExtraFiles)
 	for i := baseFd; i < baseFd+r.preserveFDs; i++ {
+		_, err := os.Stat(fmt.Sprintf("/dev/fd/%d", i))
+		if err != nil {
+			r.destroy()
+			return -1, err
```

Comment:
It'd be nice if you could wrap the error to explain what's going on -- a user isn't going to understand what `stat /proc/self/fd/150: no such file or directory` means. Something like:

```
return -1, errors.Wrap(err, "check that all preserve-fds are present")
```

Reply:
+1 Thanks.